### PR TITLE
Add support for defining parameters on scripts

### PR DESCRIPTION
**Pull request by bbbco**
Some of our scripts take extra parameters. It is nice to be able to know the ones for which this is the case.

**_magic-cli-example**_

```
#!/usr/bin/env ruby
#Just a simple example command.
# @param <random> - Random Parameter 1
# @param - Random Parameter 2

puts "It worked! 🎩✨" + ARGV.join(" ")
```

**Output**

```
╰─ ./magic-cli
usage: magic-cli <command> [<args>]

Commands:
   example <random> <param>   Just a simple example command.
   update                     Installs the latest version of these tools.
```

### CODE REVIEW

File: magic-cli [hunk 1]
```diff
@@ -43,6 +44,20 @@ class MagicCli
         return description
     end
 
+    def get_parameters(filename)
+      # Optionally, retrieve parameters that are required
+      lines = File.readlines(filename)
+      parameters = lines.map do |line|
```
Comment (shinypb) [inline magic-cli]: I think we want to stop iterating over the lines after we're past the initial header block. For example, if we had a file like this...

``` ruby
#!/usr/bin/env ruby
# Foo bar frobber
# @param foo

def some_method(a)
  # @param some_method_param
end
```

...we wouldn't want to show `some_method_param` in the main output.

**Review by shinypb — CHANGES_REQUESTED** [review]
I'm sorry for the delay — I was out on paternity leave for a while and this fell off my radar. This is a great addition to the tool, and I have just one bit of feedback for it. Thanks for your contribution!

**Comment by bbbco** [comment]
@shinypb I believe I addressed your feedback. Thanks so much!

**Comment by shinypb** [comment]
✨

**Pull request merged**